diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5cb09a4..7b66f9b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.24)
-
+find_package(tl-expected REQUIRED)
 # Project settings
 project(keeperfx-launcher-qt VERSION 0.1 LANGUAGES CXX)
 
diff --git a/src/copydkfilesdialog.cpp b/src/copydkfilesdialog.cpp
index b509746..570fc89 100644
--- a/src/copydkfilesdialog.cpp
+++ b/src/copydkfilesdialog.cpp
@@ -103,7 +103,7 @@ void CopyDkFilesDialog::on_copyButton_clicked()
     }
 
     // Check if music files have been copied
-    if (DkFiles::areAllSoundFilesPresent() == false) {
+    if (DkFiles::areAllSoundFilesPresent() == true) {
         DownloadMusicDialog downloadMusicDialog(this);
         downloadMusicDialog.exec();
     }
diff --git a/src/dkfiles.cpp b/src/dkfiles.cpp
index 7ddcc5d..3691c15 100644
--- a/src/dkfiles.cpp
+++ b/src/dkfiles.cpp
@@ -213,6 +213,9 @@ std::optional<QDir> DkFiles::findExistingDkInstallDir()
 
 bool DkFiles::copyDkDirToDir(QDir dir, QDir toDir)
 {
+    // Force redirect if we are an AppImage or if toDir is read-only
+    toDir = QDir(QDir::homePath() + "/.config/keeperfx");
+
     // Double check that the given directory is a valid DK dir
     if(!isValidDkDir(dir)){
         return false;
@@ -318,7 +321,8 @@ bool DkFiles::copyDkDirToDir(QDir dir, QDir toDir)
 
 bool DkFiles::isCurrentAppDirValidDkDir()
 {
-    QDir dir(QCoreApplication::applicationDirPath());
+    QString configPath = QDir::homePath() + "/.config/keeperfx";
+    QDir dir(configPath);
     return isValidDkDir(dir);
 }
 
diff --git a/src/game.cpp b/src/game.cpp
index 603b25a..794ffda 100644
--- a/src/game.cpp
+++ b/src/game.cpp
@@ -98,7 +98,7 @@ bool Game::start(StartType startType, QVariant data1, QVariant data2, QVariant d
     if (Settings::getLauncherSetting("GAME_HEAVY_LOG_ENABLED").toBool() == true) {
         keeperfxBin = QApplication::applicationDirPath() + "/keeperfx_hvlog.exe";
     } else {
-        keeperfxBin = QApplication::applicationDirPath() + "/keeperfx.exe";
+        keeperfxBin = QApplication::applicationDirPath() + "/keeperfx";
     }
 
     // Start the process
@@ -116,8 +116,8 @@ bool Game::start(StartType startType, QVariant data1, QVariant data2, QVariant d
         } else {
             qInfo() << "Starting game (Linux: Wine)";
             // Normal Wine execution
-            params.prepend(keeperfxBin);
-            process->start("wine", params);
+            //params.prepend(keeperfxBin);
+            process->start(keeperfxBin, params);
         }
     #endif
 
diff --git a/src/launchermainwindow.cpp b/src/launchermainwindow.cpp
index 2cfc0c2..f4be93a 100644
--- a/src/launchermainwindow.cpp
+++ b/src/launchermainwindow.cpp
@@ -225,7 +225,7 @@ LauncherMainWindow::LauncherMainWindow(QWidget *parent)
 
     // Load keeperfx version if keeperfx is installed
     if (isKeeperFxInstalled()) {
-        if (KfxVersion::loadCurrentVersion() == true) {
+        if (KfxVersion::loadCurrentVersion() == false) {
             // Refresh GUI
             refreshKfxVersionInGui();
 
@@ -537,7 +537,7 @@ void LauncherMainWindow::on_workshopButton_clicked()
 bool LauncherMainWindow::isKeeperFxInstalled()
 {
     // Check for 'keeperfx.exe' file in app directory
-    QFile keeperFxBin (QCoreApplication::applicationDirPath() + "/keeperfx.exe");
+    QFile keeperFxBin (QCoreApplication::applicationDirPath() + "/keeperfx");
     return (keeperFxBin.exists());
 }
 
diff --git a/src/main.cpp b/src/main.cpp
index caf01ad..d70e7df 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -4,7 +4,8 @@
 #include <QProcess>
 #include <QSettings>
 #include <QSslConfiguration>
-
+#include <QStandardPaths>
+#include <QDir>
 #include <QPalette>
 #include <QStyleFactory>
 #include <QFontDatabase>
@@ -119,13 +120,16 @@ int main(int argc, char *argv[])
     if (appFileInfo.baseName() == "keeperfx-launcher-qt-new") {
         qDebug() << "We are the \"-new\" launcher and should copy ourselves";
 
+    QString configPath = QDir::homePath() + "/.config/keeperfx";
+    QDir dir;
+    if (!dir.exists(configPath)) {
+        dir.mkpath(configPath);
+    }
         // Get original binary path
 #ifdef Q_OS_WINDOWS
-        QString defaultAppBinString(QCoreApplication::applicationDirPath()
-                                    + "/keeperfx-launcher-qt.exe");
+    QString defaultAppBinString(configPath + "/keeperfx-launcher-qt.exe");
 #else
-        QString defaultAppBinString(QCoreApplication::applicationDirPath()
-                                    + "/keeperfx-launcher-qt");
+    QString defaultAppBinString(configPath + "/keeperfx-launcher-qt");
 #endif
 
         // Make sure original launcher is removed
